BedrockPlugin.cpp

```diff
@@ -0,0 +1,44 @@
+#include <libstuff/libstuff.h>
+#include "BedrockPlugin.h"
+
+// Global static values
+list<BedrockPlugin*> BedrockPlugin::g_registeredPluginList;
+
+BedrockPlugin::BedrockPlugin() : _enabled(false) {
+    /* Auto-register this instance into the global static list. This just makes it available for enabling via the
```

Comment:
Can we please keep C++ comments?  By having the code use C++ comments everywhere (ie, `//`) then you can quickly block out large blocks of code using C comments (ie, `/*`).

Furthermore, I think it looks ugly to have a mix of C and C++ style comments in the code.